### DISCUSSION TITLE
How does poetry map to the correct virtual environment when there are two projects with the same name?

### DISCUSSION THREAD

**Discussion by wi0lono** [python-poetry/poetry#8865] [Q&A] [answered]
Poetry stores project specific venvs Inside the `pypoetry/virtualenvs/` directory, in the format `<project_name>-<uuid>-<python_version>`. 
I'm assuming the uuid allows poetry to differentiate between two projects having the same name in different locations.

How does it do this though? I can't see any uuid in the `pyproject.toml` file, so I'm curious as to how the correct venv is selected when running say, `poetry run python main.py`

**Comment by radoering** [comment, marked as answer]
It calculates a hash from the project's directory:

https://github.com/python-poetry/poetry/blob/7d3151631c57dbf021d3a98aca6ed5ab91e75eb9/src/poetry/utils/env/env_manager.py#L758-L765
